File: core/src/jvmMain/java/de/haukesomm/sokoban/core/moving/validation/AbstractMoveValidator.java
package de.haukesomm.sokoban.core.moving.validation;

import de.haukesomm.sokoban.core.Direction;
import de.haukesomm.sokoban.core.Entity;
import de.haukesomm.sokoban.core.GameState;
import de.haukesomm.sokoban.core.moving.MoveValidatorStatus;

import java.util.Collection;
import java.util.HashSet;

public abstract class AbstractMoveValidator implements MoveValidator {

    protected Collection<MoveValidatorStatus> singleResult(MoveValidatorStatus result) {
        var resultSet = new HashSet<MoveValidatorStatus>();
        resultSet.add(result);
        return resultSet;
    }

    @Override
    public abstract Collection<MoveValidatorStatus> check(GameState state, Entity entity, Direction direction);
}

File: core/src/commonMain/kotlin/de/haukesomm/sokoban/core/moving/validation/AbstractMoveValidator.kt
package de.haukesomm.sokoban.core.moving.validation

import de.haukesomm.sokoban.core.Direction
import de.haukesomm.sokoban.core.Entity
import de.haukesomm.sokoban.core.state.GameState

abstract class AbstractMoveValidator : MoveValidator {

    protected fun singleResult(result: MoveValidatorStatus): Collection<MoveValidatorStatus> =
        setOf(result)

    abstract override fun check(state: GameState, entity: Entity, direction: Direction): Collection<MoveValidatorStatus>
}
